### ISSUE TITLE
[backend] Pipelines/runs fail consistently at 22nd task

### ISSUE THREAD

**Issue by samoturk** [kubeflow/pipelines#9878]
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)?
Manifests
*  KFP version: 
2.0.0
*  KFP SDK version: 
2.0.1

### Steps to reproduce

Running a pipeline that has more than 22 tasks. Can be completely sequential or branched. Minimal example to reproduce:
```python
from utils.auth_session import get_istio_auth_session
from kfp.client import Client
from kfp import dsl, compiler
from kfp.dsl import Input, Output, Dataset, Model
import os
import datetime
​
COMPONENTS_IMAGE = 'alpine:3.18.2'
​
auth_session = get_istio_auth_session(
    url=os.environ['KUBEFLOW_ENDPOINT'],
    username=os.environ['KUBEFLOW_USERNAME'],
    password=os.environ['KUBEFLOW_PASSWORD']
)
​
namespace = os.environ.get('KUBEFLOW_NAMESPACE', None) or \
            os.environ['KUBEFLOW_USERNAME'].split("@")[0].replace(".", "-")
​
@dsl.container_component
def task00(output: Output[Dataset]):
    return dsl.ContainerSpec(
        image=COMPONENTS_IMAGE,
        command=['sh', '-c', 'echo hello >> $0'],
        args=[output.path])
​
@dsl.container_component
def task01(dataset: Input[Dataset], output: Output[Dataset]):
    return dsl.ContainerSpec(
        image=COMPONENTS_IMAGE,
        command=['sh', '-c', 'cp -a $0 $1'],
        args=[dataset.path, output.path])
​
​
@dsl.pipeline
def debug_containerized_pipeline():
    result00 = task00()
    result01 = task01(dataset=result00.output)
    result02 = task01(dataset=result01.output)
    result03 = task01(dataset=result02.output)
    result04 = task01(dataset=result03.output)
    result05 = task01(dataset=result04.output)
    result06 = task01(dataset=result05.output)
    result07 = task01(dataset=result06.output)
    result08 = task01(dataset=result07.output)
    result09 = task01(dataset=result08.output)
    result10 = task01(dataset=result09.output)
    result11 = task01(dataset=result10.output)
    result12 = task01(dataset=result11.output)
    result13 = task01(dataset=result12.output)
    result14 = task01(dataset=result13.output)
    result15 = task01(dataset=result14.output)
    result16 = task01(dataset=result15.output)
    result17 = task01(dataset=result16.output)
    result18 = task01(dataset=result17.output)
    result19 = task01(dataset=result18.output)
    result20 = task01(dataset=result19.output)
    result21 = task01(dataset=result20.output)
    result22 = task01(dataset=result21.output)
​
​
​
if __name__ == "__main__":
    client = Client(host=f"{os.environ['KUBEFLOW_ENDPOINT']}/pipeline", namespace=namespace,
                    cookies=auth_session["session_cookie"], verify_ssl=False)
​
    compiler.Compiler().compile(debug_containerized_pipeline, 'pipeline-debug.yaml')
​
    run = client.create_run_from_pipeline_package(
        'pipeline-debug.yaml',
        enable_caching=False,
        arguments={
        },
        run_name=f'Debug simple {datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")}'
    )
```

### Expected result

Pipelines to run beyond 22nd task.

### Materials and Reference

In the frontend the pipeline simply fails, but there are no logs of the pod that actually failed. Looking into loki reveals that the pod raises some error before our code actually runs: `F0815 13:04:52.601362      34 main.go:76] KFP driver: driver.Container(pipelineName=debug-containerized-pipeline, runID=fd3db286-a450-4b57-b66e-8dd11fbcc1e8, task="task01-21", component="comp-task01-21", dagExecutionID=1359, componentSpec) failed: failed to resolve inputs: failed to resolve input artifact dataset with spec task_output_artifact:{producer_task:"task01-20" output_artifact_key:"output"}: cannot find producer task "task01-20"`

---

Impacted by this bug? Give it a 👍.

**Comment by chensun** [comment]
@samoturk did you check your cluster's resource usage? Is that possible it run out of memory, etc.?

**Comment by samoturk** [comment]
No the cluster didn't run out of memory. I reproduced this error on two cluster, a smaller one with 64GB and a one with >1TB RAM. The attached example uses alpine Linux image which is quite frugal and that pipeline runs pods sequentially. The error nonetheless appears at exactly the same step on both clusters. Same happens with a more complex pipeline using Debian derived images - always 22nd step.

**Comment by hellojunha** [comment]
Getting executions of single DAG returns 20 executions. Querying to MLMD server limits to 20 responses in single time. This is defined in external library ml-metadata (struct `ListOperationOptions`).
I got the exactly same problem and solved by increasing MaxResultSize 100 at `GetExecutionsInDAG` of backend/src/v2/metadata/client.go https://github.com/kubeflow/pipelines/blob/1cc31bb9179e936efd3f448975e6631c558c2e15/backend/src/v2/metadata/client.go#L705-L707

I think there should be something fetches next page if exist..

**Comment by geier** [comment]
For us, this was fixed with #10396.
